Src/Plugins/ValidatorBase.cs

```diff
@@ -144,6 +146,12 @@ public static string ReturnUnknownHost(ref string message, string host)
             return nameof(ValidationState.Unknown);
         }
 
+        public static string ParseExpression(IRegex regexEngine, string matchedPattern, string expression)
```

Comment:
The two methods added here serve almost the exact same function as the two i added to the PostgeSQL validator.  Could you either consolidate the two, or create a TODO or work item to do so?

Reply:
So, by the way, it turns out that RE2 supports groups, which we were confused about early on (and so didn't plumb the support through). If we exposed this, we wouldn't need to do this parse value stuff. Anyone interested in fixing up RE2 to close this gap? Let me know.

---
In reply to: [569657561](https://github.com/microsoft/sarif-pattern-matcher/pull/160#discussion_r569657561) [](ancestors = 569657561)

Reply:
where should we fix that? if u could point me i could take a look :)


---
In reply to: [569695240](https://github.com/microsoft/sarif-pattern-matcher/pull/160#discussion_r569695240) [](ancestors = 569695240,569657561)

Reply:
I'll do it in my branch, disregard!

---
In reply to: [569696512](https://github.com/microsoft/sarif-pattern-matcher/pull/160#discussion_r569696512) [](ancestors = 569696512,569695240,569657561)